### ISSUE TITLE
Lack of clarity in Cilium-Istio integration documentation

### ISSUE THREAD

**Issue by psinghal04** [cilium/cilium#15352]
The page [Getting Started Using Istio](https://docs.cilium.io/en/v1.9/gettingstarted/istio/)  states the following in the second paragraph:

_"Cilium’s Istio integration allows Cilium to enforce HTTP L7 network policies within the Istio sidecar proxies. Note that Istio can also be deployed without Cilium integration by running a standard version of istioctl. In that case Cilium will enforce HTTP L7 policies outside of the Istio sidecar proxy."_

So, when should one consider enforcing HTTP L7 network policies within the Istio sidecar proxies, versus outside (in other words, when should one consider deploying Cilium's version of Istio)? Perhaps a couple of lines addressing that might help enhance the context of this page. 

The page describes several examples of using CiliumNetworkPolicy in an Istio deployment. If I am using the Cilium CNI plugin in my cluster, what difference in behavior can I expect by using the standard Istio version vs. the Cilium version of Istio?

Are the benefits of using Cilium with Istio highlighted in the following Blog post achieved even with the standard Istio version, or do we need to use the Cilium version of Istio for some of them?  https://cilium.io/blog/2018/08/07/istio-10-cilium

Some more detail and clarity on the value-add of the Cilium's version of Istio would be very helpful!

**Comment by pchaigno** [comment]
> If I am using the Cilium CNI plugin in my cluster, what difference in behavior can I expect by using the standard Istio version vs. the Cilium version of Istio?

AFAIK, only the Cilium version of Istio can enforce L7 policies and it's packaged with Cilium. I think the sentence `In that case Cilium will enforce HTTP L7 policies outside of the Istio sidecar proxy` refers to the fact that the standard Istio installed by users won't be used to enforce L7 policies; the Cilium Istio will be used instead.

> Are the benefits of using Cilium with Istio highlighted in the following Blog post achieved even with the standard Istio version, or do we need to use the Cilium version of Istio for some of them? https://cilium.io/blog/2018/08/07/istio-10-cilium

I don't think the socket-level redirection is specific to Istio. It should work for any pod-to-pod connection on the same node.

**Comment by psinghal04** [comment]
@pchaigno, I got a chance to engage with @tgraf on this question. If I have understood his response correctly, the use-case for using Cilium's downstream version of Istio is when you intend to use Cilium's L7 network policies in a mesh that is set up for mTLS, and those policies contain rules around metadata (such as HTTP headers) that would otherwise appear encrypted to Cilium if it were enforcing the policy from the "outside". When using Cilium's downstream version of Istio, these network policies are enforced by Cilium within the sidecar proxies - which allows it to enforce them to see the metadata unencrypted, and apply the L7 network policies correctly. Beyond that, as you said, the socket-level redirection (and all of it's associated benefits) are available for any distribution of Istio, including upstream Istio.

Hopefully I got it right, but if not, would very much appreciate further clarification. Thanks!

**Comment by psinghal04** [comment]
@pchaigno , after looking into this a bit, I think I need some more clarification. I came to understand that the use-case of using Cilium's downstream version of Istio was if we needed to enforce L7 policies via CiliumNetworkPolicy, where we need policy to be enforced within the sidecar proxy so that it had access to unencrypted HTTP headers (in case of mTLS).  

However, it seems Istio's [AuthorizationPolicy](https://istio.io/latest/docs/reference/config/security/authorization-policy/) resource offers the same L7 policy features as CiliumNetworkPolicy, and the AuthorizationPolicy would be enforced within the Envoy proxy with standard upstream Istio. 

So, once again - I am left wondering - what exactly is the use-case where we might need Cilium's Istio distribution?
